src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -123,6 +123,77 @@ public abstract partial class Compilation
 
         internal abstract void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder);
 
+        /// <summary>
+        /// This method generates a string that represents the input content to the compiler which impacts 
+        /// the output of the build. This string is effectively a content key for a <see cref="Compilation"/>
+        /// with these values that can be used to identify the outputs.
+        ///
+        /// The returned string has the following properties:
+        /// - The format is undefined. Consumers should assume the format and content can change between 
+        ///   compiler versions.
+        /// - It is designed to be human readable and diffable. This is to help developers
+        ///   understand the difference between two compilations which is impacting the deterministic 
+        ///   output
+        /// - It is *not* in a minimal form. If used as a key in say a content addressable storage consumers
+        ///   should first pass it through a strong hashing function.
```

Comment:
fwiw, this is not going to show up properly in QuickInfo.  Need things like `<list>` and `<item>` unfortunately.

Reply:
May as well do it right :smile: